crossdock/client/gauntlet/behavior.go

```diff
@@ -0,0 +1,485 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package gauntlet
+
+import (
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/yarpc/yarpc-go/crossdock/client/behavior"
+	"github.com/yarpc/yarpc-go/crossdock/client/params"
+	"github.com/yarpc/yarpc-go/crossdock/client/random"
+	"github.com/yarpc/yarpc-go/crossdock/thrift/gauntlet"
+	"github.com/yarpc/yarpc-go/crossdock/thrift/gauntlet/yarpc/secondserviceclient"
+	"github.com/yarpc/yarpc-go/crossdock/thrift/gauntlet/yarpc/thrifttestclient"
+	"github.com/yarpc/yarpc-go/encoding/thrift"
+	"github.com/yarpc/yarpc-go/transport"
+
+	"golang.org/x/net/context"
+)
+
+type gauntletEntry struct {
+	behavior.Entry
+
+	Transport string `json:"transport"`
+	Server    string `json:"server"`
+}
+
+type gauntletSink struct {
+	behavior.Sink
+
+	Transport string
+	Server    string
+}
+
+func (s gauntletSink) Put(e interface{}) {
+	s.Sink.Put(gauntletEntry{
+		Entry:     e.(behavior.Entry),
+		Transport: s.Transport,
+		Server:    s.Server,
+	})
+}
+
+func createGauntletSink(s behavior.Sink, ps behavior.Params) behavior.Sink {
+	return gauntletSink{
+		Sink:      s,
+		Transport: ps.Param(params.Transport),
+		Server:    ps.Param(params.Server),
+	}
+}
+
+// Run executes the thriftgauntlet behavior.
+func Run(s behavior.Sink, ps behavior.Params) {
+	s = createGauntletSink(s, ps)
+	assert := behavior.Assert(s)
+	checks := behavior.Checks(s)
+
+	rpc := behavior.CreateRPC(s, ps)
+	channel := rpc.Channel("yarpc-test")
+
+	bytesToken := random.Bytes(10)
+	tests := []struct {
+		service  string        // thrift service name; defaults to ThriftTest
+		function string        // name of the Go function on the client
+		details  string        // optional extra details about what this test does
+		give     []interface{} // arguments besides thrift.Request
+
+		want          interface{} // expected response; nil for void
+		wantError     error       // expected error
+		wantErrorLike string      // for just matching error messages
+	}{
+		{
+			function: "TestBinary",
```

Comment:
Update: Added tests for the other service. New clients need to be instantiated per-test.